EventHubs/tests/baseeventhubtest.py

```diff
@@ -92,8 +50,89 @@ def _parse_template(self):
+        )
+        
+        with producer:
+            try:
+                producer.send_batch(event_data_list)
+            except ValueError:  # Size exceeds limit. This shouldn't happen if you make sure before hand.
+                print("Size of the event data list exceeds the size limit of a single send")
+            except EventHubError as eh_err:
+                print("Sending error: ", eh_err)
+
+        print("Event inserted")
+    
+    def fetchlogs(self, app_insights):
+        result = []
+        try:
+            client = LogsQueryClient(self.azure_credential)
+            query = f"app('{app_insights}').traces | where operation_Name == '{self.function_name}' | project operation_Id, timestamp, message, severityLevel"
+            response = client.query_workspace(self.get_Workspace_Id(), query, timespan=timedelta(hours=1))
+                
+            if response.status == LogsQueryStatus.FAILURE:
+                raise Exception(f"LogsQueryError: {response.message}")
+            elif response.status == LogsQueryStatus.PARTIAL:
+                data = response.partial_data
+                error = response.partial_error
+                print("partial_error: ", error)
+            elif response.status == LogsQueryStatus.SUCCESS:
+                data = response.tables
+
+            for table in data:
+                for row in table.rows:
+                    row_dict = {str(col): str(item) for col, item in zip(table.columns, row)}
+                    result.append(row_dict)
+        except Exception as e:
+            print("An unexpected error occurred during the test:")
+            print("Exception", e)
+
+        return result
+    
+    def get_resources(self, resource_group_name):
+        return self.resource_client.resources.list_by_resource_group(resource_group_name)
+    
+    def get_Workspace_Id(self):
+        workspace = self.get_resource('microsoft.operationalinsights/workspaces')
+        client = LogAnalyticsManagementClient(
+            credential=self.azure_credential,
+            subscription_id=self.subscription_id,
+        )
+
+        response = client.workspaces.get(
+            resource_group_name=self.RESOURCE_GROUP_NAME,
+            workspace_name=workspace.name,
+        )
+        return response.customer_id
+    
+    def filter_logs(self, logs, key, value):
+        return value in [d.get(key) for d in logs]
+    
+    def check_resource_count(self):
+        resource_count = len(list(self.get_resources(self.RESOURCE_GROUP_NAME)))
+        self.assertTrue(resource_count == self.expected_resource_count, f"resource count of resource group {self.RESOURCE_GROUP_NAME} differs from expected count : {resource_count}")
+
+    def check_success_log(self, logs):
+        self.assertTrue(self.filter_logs(logs, 'message', self.successful_sent_message))
+    
+    def check_error_log(self, logs):
+        self.assertTrue(not self.filter_logs(logs, 'severityLevel', '3'))
```

Comment:
may be we can assertFalse

Reply:
add messages otherwise we don't know what failed
self.asserTrue(self.filter_logs(logs, 'message', self.successful_sent_message), "No success message found in azure function logs")
add messages self.asserFalse(self.filter_logs(logs, 'severityLevel', '3'), "Error  messages found in azure function logs")
add messages self.asserFalse(self.filter_logs(logs, 'severityLevel', '2'), "Warning messages found in azure function logs")